go/staking/api/slashing.go

```diff
@@ -41,6 +41,10 @@ func (s SlashReason) MarshalText() ([]byte, error) {
 // UnmarshalText decodes a text slice into a SlashReason.
 func (s *SlashReason) UnmarshalText(text []byte) error {
 	switch string(text) {
+	// XXX: The "0" case is only for backward compatibility, so that the old
+	// genesis file loads -- remove this once mainnet is upgraded!
```

Comment:
Wouldn't it be better if we handled this in fixgenesis?

Reply:
No, because there are too many structures to duplicate and there's more chance of something going wrong with the conversion (and since this is in staking code it's better to be more careful).  A two-line fix like this is cleaner, IMO :)
I'll file an issue later, so we don't forget to remove this once the upgrade is complete.

Reply:
Filed as https://github.com/oasisprotocol/oasis-core/issues/3627 :)